Sources/SwiftWarningControl/SwiftWarningControl.md

```diff
@@ -0,0 +1,30 @@
+# SwiftWarningControl
+
+A library to evaluate `@warn` diagnostic group controls within a Swift syntax tree.
+
+## Overview
+
+Swift provides a mechanism to control the behavior of specific diagnostic groups for a given declaration's lexical scope with the `@warn` attribute.
+
+The syntax tree and its parser do not reason about warning group controls. The syntax tree produced by the parser represents the `@warn` attribute in a generic fashion, as it would any other basic attribute on a declaration. The per-declaration nature of the attribute means that for any given lexical scope, the behavior of a given diagnostic group can be queried by checking for the presence of this attribute in its parent declaration scope. 
+
+```swift
+@warn(Deprecate, as: error)
+func foo() {
+    ...
+    @warn(Deprecate, as: warning)
+    func bar() {
+        ...
+        @warn("Deprecate", as: ignored, reason: "Foo")
+        func baz() {
+          ...
+        }
+    }
+}
```

Comment:
Another nitpick: Would you indent this by two spaces to match the rest of the repo?